### PR TITLE
feat(wallet) Add warning in Send Screen for BTC transactions

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/35514

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://github.com/brave/brave-core/assets/48117473/5a3dd3e0-eec0-46e2-9fc9-1cef0c526418

### CODE REVIEW

File: components/brave_wallet_ui/page/screens/send/components/ordinals-warning-message/ordinals-warning-message.style.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  width: 100%;
+  padding: ${leo.spacing.m};
+  gap: 10px;
+  background-color: ${leo.color.systemfeedback.warningBackground};
+  border-radius: ${leo.radius.m};
+  margin-top: ${leo.spacing.xl};
+  margin-bottom: ${leo.spacing.xl};
+`
+
+export const WarningIcon = styled(Icon)`
```
Comment (josheleonard) [inline components/brave_wallet_ui/page/screens/send/components/ordinals-warning-message/ordinals-warning-message.style.ts:27]: How about using `attrs` to lock-in which icon to display?

```suggestion
export const WarningIcon = styled(Icon).attrs({
  name: 'warning-triangle-filled
})`
```

**Review by josheleonard — APPROVED** [review]
++

**Review by nvonpentz — APPROVED** [review]

**Review by mkarolin — APPROVED** [review]
`strings`++

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.64.30 Chromium: 121.0.6167.85 (Official Build) nightly (64-bit)
-- | --
Revision | 01fbd61f11c42b0844bd87fe85ebfb6ab2615bd3
OS | Linux

- Verified steps from PR description
- Verified warning is shown in Send screen for BTC transactions on both Mainnet and Testnet

https://github.com/brave/brave-core/assets/17010094/a711a22a-b7a3-4bb6-96c9-388a4d7b465d

**Pull request merged**